atef/check.py

```diff
@@ -0,0 +1,707 @@
+from __future__ import annotations
+
+import enum
+import logging
+from dataclasses import dataclass, field
+from typing import (Any, Callable, ClassVar, Dict, Generator, List, Optional,
+                    Sequence, Tuple, Type, Union)
+
+import numpy as np
+import ophyd
+
+from . import serialization
+
+Number = Union[int, float]
+
+logger = logging.getLogger(__name__)
+
+
+class Severity(enum.IntEnum):
+    success = 0
+    warning = 1
+    error = 2
+    internal_error = 3
```

Comment:
I assume this is differentiating between "my device has an issue" and "my test has an issue", really good distinction

Reply:
Yes, exactly. I have notes elsewhere here about not knowing where "OK for beam" and "device seems OK" fits in, though.

Reply:
My two immediate thought threads on this are basically:
- Can we simply tag tests as "beam tests" and "device tests"?
- Can we include some sort of "consequence" enum in the failure information of a test?

not 100% sure here

Reply:
Hmm, I think the beam vs device test may be a good way to separate it out... I think we'll need to whiteboard this to really understand the implications of it all though.